src/server/rdb_load.cc

```diff
@@ -2348,6 +2388,38 @@ error_code RdbLoaderBase::HandleJournalBlob(Service* service) {
   return std::error_code{};
 }
 
+error_code RdbLoader::LoadTieredPage() {
+  size_t offset;
+  SET_OR_RETURN(LoadLen(nullptr), offset);
+
+  std::string page;
+  SET_OR_RETURN(FetchGenericString(), page);
+
+  // If tiering is enabled, try saving the received page on disk
+  // Fall back to memory in case of errors
+  if (EngineShard::tlocal() && EngineShard::tlocal()->tiered_storage()) {
+    auto& storage = EngineShard::tlocal()->tiered_storage()->BorrowStorage();
+
+    util::fb2::Done done;
+    std::error_code ec;
+    auto cb = [this, offset, &ec, &done](io::Result<tiering::DiskSegment> res) {
+      if (res.has_value())
+        small_items_pages_[offset] = res.value();
+      else
+        ec = res.error();
+      done.Notify();
+    };
+    ec = storage.Stash(io::Buffer(page), {}, cb);
+
+    done.Wait();
+    if (!ec)
+      return {};
+  }
+
+  small_items_pages_[offset] = page;
+  return {};
+}
```

Comment:
I really wish to use "virtual" keys to store pages. This would not require having custom code for their serialization, as well custom code for loading and stashing them. We could simply rely on the tiered storage for doing this

Reply:
Second, a missing optimization is that we don't re-use the offloaded page, we only use it to read from it

Reply:
The idea would be to create a mapping table between new and old offsets and re-create those offsets in small bins

Reply:
you can add TODOs to detail your ideas..
Regarding virtual keys, I think it's fine to use a predefined prefix like `__df__` for dragonfly specific keys if it helps you. one thing to note - how it works with multiple databases. pages are shared between entities from differrent databases,imho

Reply:
>  if it helps you

I want to use the existing code for serialization and offloading, so pages will be "just values" and managed automatically